cmd/dra-example-kubeletplugin/state.go

```diff
@@ -157,11 +167,38 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 			return nil, fmt.Errorf("requested GPU is not allocatable: %v", result.Device)
 		}
 
-		device := &drapbv1.Device{
-			RequestNames: []string{result.Request},
-			PoolName:     result.Pool,
-			DeviceName:   result.Device,
-			CDIDeviceIDs: s.cdi.GetClaimDevices([]string{result.Device}),
+		config, err := GetMergedOpaqueDeviceConfig[configapi.GpuConfig](
+			configapi.Decoder,
+			DriverName,
+			result.Request,
+			claim,
+			deviceClasses,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error getting merged GPU config: %v", err)
+		}
+
+		if config.Sharing == nil {
+			config.Sharing = configapi.DefaultGpuSharingConfig()
+		}
+
+		device := &PreparedDevice{
+			Device: drapbv1.Device{
+				RequestNames: []string{result.Request},
+				PoolName:     result.Pool,
+				DeviceName:   result.Device,
+				CDIDeviceIDs: s.cdi.GetClaimDevices([]string{result.Device}),
+			},
+			Config: config,
+		}
+
+		// Apply any requested configuration here.
```

Comment:
What about validation?

You have used `+kubebuilder:validation` above, but that doesn't seem to be used for anything.

Reply:
Manual validation code has now been added.